app/helpers/fenrir_view/component_helper.rb

```diff
@@ -1,23 +1,63 @@
+# frozen_string_literal: true
+
 module FenrirView
   module ComponentHelper
+
+    # TODO: Remove?
     def render_ui(variant, slug, properties = {}, &block)
       return nil if properties == false
 
       component = FenrirView::Presenter.component_for(variant, slug, properties, validate: true)
+
+      if block_given?
+        if defined?(component.layout)
+          component.layout = self
+          capture { yield component.layout }
+        else
+          content_block = capture(&block)
```

Comment:
Could use the same yield syntax as above to make it more obvious what the difference is (passing the argument 'layout' to the block)